update_devfile_patched_image_tags.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+set -e
+
+DEFAULT_TAG="nightly"
+DEFAULT_REGISTRY="quay.io"
+DEFAULT_ORGANIZATION="eclipse"
+
+TAG=${TAG:-$DEFAULT_TAG}
+REGISTRY=${REGISTRY:-$DEFAULT_REGISTRY}
+ORGANIZATION=${ORGANIZATION:-$DEFAULT_ORGANIZATION}
+
+if [ "$TAG" = "$DEFAULT_TAG" ] && [ "$ORGANIZATION" = "$DEFAULT_ORGANIZATION" ] && [ "$REGISTRY" = "$DEFAULT_REGISTRY" ]; then
```

Comment:
what happens if TAG = GIT Commit hash. Wondering about `master` CI

Reply:
I'm not sure `TAG` is ever set to the commit hash. In the CI scripts, I only see it set to either `nightly` or `head -n 1 VERSION`. We do tagging based off the git commit as well, but it is just a [tag on the built image](https://github.com/eclipse/che-devfile-registry/pull/90/files#diff-3961cf6dd52815a90fe783caf9af804dR103) and is only done for regular builds (i.e. ones where we don't change devfiles). 

Note that `update_devfile_patched_image_tags.sh` will only execute when `docker build` is called with `--build-arg PATCHED_IMAGES_[REG|ORG|TAG]=<non-default>`.

Reply:
TAG based on the commit hash is set as part of every CI https://github.com/eclipse/che-devfile-registry/blob/master/cico_functions.sh#L100-101 but I guess it will not be called for regular CI case since --build-arg are not used here - https://github.com/eclipse/che-devfile-registry/blob/master/cico_functions.sh#L116-L117

Reply:
To add, the tag based on commit has *is* set, but it uses the env var [`${GIT_COMMIT_TAG}`](https://github.com/eclipse/che-devfile-registry/blob/53d89c445b47be6cb2431ca34ebbec5df4cba377/cico_functions.sh#L68), not `${TAG}`, so even in the release case the git commit hash would not be used.